src/core/amount.js

```diff
@@ -743,8 +762,9 @@ Amount.prototype.to_text = function() {
   // not native
   const offset = this._value.getExponent() - 15;
   const sign = this._value.isNegative() ? '-' : '';
-  const mantissa = utils.getMantissa16FromString(
-    this._value.abs().toString());
+  const mantissa = this._value.isNegative() ?
```

Comment:
So consider that if I come back to this in a few months, I might refactor it back to the way it was before. I think it's worth making a helper function called `abs` to do this and giving it a comment explaining that the BigNumber constructor is very slow.